src/event.lib.php

```diff
@@ -752,12 +757,74 @@ public static function setNewPassword($user_id = 0)
             $data_set
         );
 
-        /* send email before changing the password */
-        wp_set_password($new_password, $user_id);
-
         return true;
     }
 
+    /**
+     * Gets a new password reset key.
+     *
+     * @since 1.8.25
+     *
+     * @param WP_User $user WP_User object.
+     * @return string|false Returns a password reset key as a string, false otherwise.
+     */
+    public static function GetPasswordResetKey($user)
+    {
+        global $wp_hasher;
+
+        if (!($user instanceof WP_User)) {
+            return false;
+        }
+
+        /**
+         * As of version 1.8.25 of this plugin, we still support WordPress version 3.6 and up
+         * and for that reason we can't take advantage of the native function get_password_reset_key
+         * (https://developer.wordpress.org/reference/functions/get_password_reset_key/), introduced in
+         * WordPress 4.4.
+         *
+         * When we drop support for versions prior to WordPress 4.4, we can use get_password_reset_key
+         * instead of this function.
```

Comment:
Given that this function is going to be replaced by the native one, it makes sense these functions to be interchangeable. That means the return type should be `string|WP_Error`, instead of `string|bool` and the errors be handled by the caller. What do you think?